glue/viewers/table/qt/tests/test_data_viewer.py

```diff
@@ -460,6 +460,49 @@ def test_incompatible_subset():
     assert refresh2.call_count == 0
 
 
+def test_table_incompatible_attribute():
+    """
+    Regression test for a bug where the table viewer generates an
+    uncaught IncompatibleAttribute error if one toggles the
+    visibility of a dataset iff an invalid subset exists. This
+    occurred because disabled layer_artists were still showing
+    as visible.
+    """
+    app = get_qapp()
+    d1 = Data(x=[1,2,3,4],y=[5,6,7,8],label='d1')
+    d2 = Data(a=['a','b','c'],b=['x','y','z'],label='d2')
+    dc = DataCollection([d1,d2])
+    gapp = GlueApplication(dc)
+    viewer = gapp.new_data_viewer(TableViewer)
+    viewer.add_data(d2)
+
+    #This subset should not be shown in the viewer
```

Comment:
```suggestion
```suggestion
    d1 = Data(x=[1, 2, 3, 4], y=[5, 6, 7, 8], label='d1')
    d2 = Data(a=['a', 'b', 'c'], b=['x', 'y', 'z'], label='d2')
    dc = DataCollection([d1, d2])
    gapp = GlueApplication(dc)
    viewer = gapp.new_data_viewer(TableViewer)
    viewer.add_data(d2)

    # This subset should not be shown in the viewer
```
Here and elsewhere, see coding guidelines in the developer guide, specifically the [style conventions](http://docs.astropy.org/en/stable/development/codeguide.html#coding-style-conventions); those were the E231 and E265 failures in the codestyle test.

Reply:
I've resolved the issue with my codestyle setup.